### PR TITLE
Improve docs and correct errors

### PR DESCRIPTION
**Pull request by austinzheng**
No functional change.

### CODE REVIEW

File: Realm/RLMSyncManager.h [hunk 1]
```diff
@@ -62,9 +62,13 @@ typedef void(^RLMSyncErrorReportingBlock)(NSError *, RLMSyncSession * _Nullable)
 @interface RLMSyncManager : NSObject
 
 /**
- An optional block which can be used to report sync-related errors to your application. Errors reported through this
- mechanism are always fatal; they represent attempts to open sessions which are invalid (for example, using malformed
- URLs).
+ A block which can optionally be used to report sync-related errors to your application.
```
Comment (jpsim) [inline Realm/RLMSyncManager.h]: It can't optionally be used. It can optionally be _set_.

File: Realm/RLMSyncManager.h [hunk 2]
```diff
@@ -62,9 +62,13 @@ typedef void(^RLMSyncErrorReportingBlock)(NSError *, RLMSyncSession * _Nullable)
...
+
+ Errors reported through this mechanism are fatal, with several exceptions. Please consult
+ `RLMSyncError` for information about the types of errors that can be reported through
+ the block, and for information on handling the error types that are recoverable.
```
Comment (jpsim) [inline Realm/RLMSyncManager.h]: How's this?

> for suggestions on handling recoverable error codes.

File: Realm/RLMSyncManager.h [hunk 3]
```diff
@@ -83,8 +87,13 @@ typedef void(^RLMSyncErrorReportingBlock)(NSError *, RLMSyncSession * _Nullable)
 @property (nonatomic) BOOL disableSSLValidation;
 
 /**
- The logging threshold which newly opened synced Realms will use. Defaults to `RLMSyncLogLevelInfo`. Set this before
- any synced Realms are opened. Logging strings are output to ASL.
+ The logging threshold which newly opened synced Realms will use. Defaults to
+ `RLMSyncLogLevelInfo`.
+
+ Logging strings are output to ASL.
```
Comment (jpsim) [inline Realm/RLMSyncManager.h]: I think it's worth spelling out the acronym:

> Logs are forwarded to Apple System Logger.

File: Realm/RLMSyncSession.h [hunk 4]
```diff
@@ -99,7 +99,7 @@ NS_ASSUME_NONNULL_BEGIN
  (and a remote Realm at a given URL stored on a Realm Object Server).
 
  Sessions are always created by the SDK and vended out through various APIs. The lifespans
- of sessions associated with Realms are managed automatically.
+ of sessions associated with Realms are managed automatically. Sessions are thread safe.
```
Comment (jpsim) [inline Realm/RLMSyncSession.h]: Thread safe is vague.

> Session objects can be accessed from any thread.

File: Realm/RLMSyncUser.h [hunk 5]
```diff
@@ -38,17 +38,20 @@ typedef void(^RLMUserCompletionBlock)(RLMSyncUser * _Nullable, NSError * _Nullab
 NS_ASSUME_NONNULL_BEGIN
 
 /**
- A `RLMSyncUser` instance represents a single Realm Object Server user account (or just user).
+ A `RLMSyncUser` instance represents a single Realm Object Server user account
+ (or just user).
```
Comment (jpsim) [inline Realm/RLMSyncUser.h]: I don't get what the parentheses mean. How is "just user" a different thing than what's stated before ("a single Realm Object Server user account")?

File: Realm/RLMSyncUser.h [hunk 6]
```diff
@@ -38,17 +38,20 @@ typedef void(^RLMUserCompletionBlock)(RLMSyncUser * _Nullable, NSError * _Nullab
...
 
- A user may have one or more credentials associated with it. These credentials uniquely identify the user to a
- third-party auth provider, and are used to sign into a Realm Object Server user account.
+ A user may have one or more credentials associated with it. These credentials
+ uniquely identify the user to a third-party auth provider, and are used to sign
```
Comment (jpsim) [inline Realm/RLMSyncUser.h]: there's no guarantee that the authentication provider is from a third party.

> uniquely identify the user to the authentication provider

File: Realm/RLMSyncUser.h [hunk 7]
```diff
@@ -38,17 +38,20 @@ typedef void(^RLMUserCompletionBlock)(RLMSyncUser * _Nullable, NSError * _Nullab
...
+ into a Realm Object Server user account.
 
- Note that users are only vended out via SDK APIs, and only one user instance ever exists for a given user account.
+ Note that user objects are only vended out via SDK APIs, and cannot be directly
+ initialized. Users are thread safe.
```
Comment (jpsim) [inline Realm/RLMSyncUser.h]: Thread safe is vague.

> User objects are safe to access from any thread.

File: Realm/RLMSyncUser.h [hunk 8]
```diff
@@ -77,6 +80,8 @@ NS_ASSUME_NONNULL_BEGIN
 
 /**
  The current state of the user.
+
+ @see `RLMSyncUserState`
```
Comment (jpsim) [inline Realm/RLMSyncUser.h]: Why this `@see`? It just points to the type this is documenting. `@see` blocks are used to refer to _other_ types.

**Comment by austinzheng** [comment]
CI failures look spurious. Merging.

**Pull request merged**